Storage/tests/Unit/ConformanceTest.php

```diff
@@ -90,13 +78,76 @@ public function testSignedUrlConformance(array $testdata)
             $testdata['bucketBoundDomain']
         );
 
-        $signedUrl = $signingObject->$instanceMethodName($expiration, $testdata + [
+        $signedUrl = $signingObject->$instanceMethodName($expiration, $hostnameOptions + $testdata + [
             'version' => 'v4'
         ]);
 
         $this->assertEquals($expectedUrl, $signedUrl);
     }
 
+    /**
+     * @group storage-conformance-post-policy
+     * @dataProvider postPolicyConformanceCases
+     */
+    public function testPostPolicy(array $testdata)
+    {
+        $testdata['policyInput'] = $testdata['policyInput'] + [
+            'conditions' => [],
+            'fields' => []
+        ];
+
+        $client = new StorageClient([
+            'keyFilePath' => __DIR__ . '/data/signed-url-v4-service-account.json'
+        ]);
+
+        $generationTimestamp = \DateTimeImmutable::createFromFormat(
+            \DateTime::RFC3339,
+            $testdata['policyInput']['timestamp']
+        );
+        $bucket = $client->bucket($testdata['policyInput']['bucket']);
+
+        $expiration = $generationTimestamp->format('U') + $testdata['policyInput']['expiration'];
+
+        $hostnameOptions = $this->getHostnameOptions($testdata['policyInput']);
+
+        $conditions = [];
+        foreach ($testdata['policyInput']['conditions'] as $key => $condition) {
+            if ($key === 'startsWith') {
+                $key = 'starts-with';
+            }
+
+            if ($key === 'contentLengthRange') {
+                $key = 'content-length-range';
+            }
+
+            if (!is_array($condition)) {
+                $condition = [$condition];
+            }
+
+            $conditions[] = array_merge([$key], $condition);
+        }
+
+        $options = [
+            'timestamp' => $generationTimestamp,
+            'fields' => $testdata['policyInput']['fields'],
+            'conditions' => $conditions,
+        ];
+
+        $policy = $bucket->generateSignedPostPolicyV4(
+            $expiration,
+            $testdata['policyInput']['object'],
+            $hostnameOptions + $options
+        );
+
+        $decodedPolicy = base64_decode($policy['fields']['policy']);
+        $this->assertEquals($testdata['policyOutput']['fields'], $policy['fields']);
+        $this->assertEquals($testdata['policyOutput']['url'], $policy['url']);
+        // $this->assertEquals(
```

Comment:
debugging code, probably should be removed or enabled?

Reply:
removed, thanks!